### PR TITLE
Update examples in vade evan

### PR DESCRIPTION
**Pull request by dimgord**
**Description**

The examples in vade-evan seem to be outdated, e.g. the TypeScript interfaces in the wasm folder do not match the interfaces of vade-sidetree.

**Details**

- update TypeScript API in WASM with latest version from ssi-core
- 
- check inputs and outputs in all examples
- 
- We can also remove the CL examples

Add a plain example for sidetree by using the vade-shared typescript class.

**Acceptance Criteria**

all examples can be started and work

### CODE REVIEW

File: builds/wasm/src/vade/vade-api-bbs.spec.ts [hunk 1]
```diff
@@ -14,7 +15,7 @@ function keyOf(did: string) {
   return `${did}#key-1`;
 }
 
-const CREDENTIAL_OFFER_TYPE = 'EvanBbsCredentialOffering';
+// const CREDENTIAL_OFFER_TYPE = 'EvanBbsCredentialOffering';
```
Comment (Tschuck) [inline builds/wasm/src/vade/vade-api-bbs.spec.ts]: Can be deleted

```suggestion
// const CREDENTIAL_OFFER_TYPE = 'EvanBbsCredentialOffering';
```

File: builds/wasm/src/vade/typings/index.ts [hunk 2]
```diff
@@ -13,88 +13,73 @@ import {
...
+  CredentialSubject as CredentialSubjectJwt,
+  IssueCredentialPayload as IssueCredentialPayloadJwt,
+  IssueCredentialResult as IssueCredentialResultJwt,
+  ProofVerification as ProofVerificationJwt,
+  SchemaProperty as SchemaPropertyJwt,
+  SignerOptions as SignerOptionsJwt,
+  TypeOptions as TypeOptionsJwt,
+  UnsignedCredential as UnsignedCredentialJwt,
+  VerifyProofPayload as VerifyProofPayloadJwt,
+} from './vade-jwt-vc';
```
Comment (Tschuck) [inline builds/wasm/src/vade/typings/index.ts:31]: I think this one is missing, cant run the tests.

Comment (S3bb1) [reply builds/wasm/src/vade/typings/index.ts:31]: This is because you havent cloned vade-jwt-vc

File: builds/wasm/src/vade/vade-api-bbs.ts [hunk 3]
```diff
@@ -48,6 +48,28 @@ class VadeApiBbs extends VadeApiShared {
...
+  context: Context,
+): Promise<CredentialSchema> {
+  checkRequiredProperties(
+    params,
+    ['identity', 'name', 'properties', 'publicKeyDidId', 'requiredProperties', 'credentialDid'],
+    'params',
+  );
+
+  checkRequiredProperties(options, ['signingKey'], 'options');
+
```
Comment (wulfraem) [inline builds/wasm/src/vade/vade-api-bbs.ts]: ? ^^

```suggestion
```

File: builds/wasm/src/vade/vade-api-bbs.ts [hunk 4]
```diff
@@ -57,6 +79,7 @@ class VadeApiBbs extends VadeApiShared {
       publicKeyDidId: string,
       requiredProperties: string[],
       description?: string,
+      credentialDid: string,
```
Comment (wulfraem) [inline builds/wasm/src/vade/vade-api-bbs.ts:61]: Missing in function doc. ^^

File: builds/wasm/src/vade/vade-api-bbs.ts [hunk 5]
```diff
@@ -111,10 +138,11 @@ class VadeApiBbs extends VadeApiShared {
   public async createMasterSecret(): Promise<string> {
     return this.executeVade<BbsTypeOptions, void, string>(
       {
-        command: 'run_custom_function',
-        customFunction: 'create_master_secret',
+        command: 'vc_zkp',
+        subcommand: 'create_master_secret',
```
Comment (wulfraem) [inline builds/wasm/src/vade/vade-api-bbs.ts]: Does this work with

```ts
    const cmd = `${command}_${subcommand}`;
```

in `vade-api-shared`?

File: builds/wasm/src/vade/vade-api-bbs.ts [hunk 6]
```diff
@@ -139,15 +167,16 @@ class VadeApiBbs extends VadeApiShared {
     );
     const ms = await this.executeVade<BbsAuthenticationOptions, CreateKeysPayload, BbsKeys>(
       {
-        command: 'run_custom_function',
-        customFunction: 'create_new_keys',
-        method: 'did:evan',
+        command: 'vc_zkp',
```
Comment (wulfraem) [inline builds/wasm/src/vade/vade-api-bbs.ts]: Does this work with

```ts
    const cmd = `${command}_${subcommand}`;
```

in `vade-api-shared`?

File: builds/wasm/src/vade/vade-api-sidetree.ts [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+export enum PublicKeyPurpose {
+  Authentication = 'authentication',
+  AssertionMethod = 'assertionMethod',
+  CapabilityInvocation = 'capabilityInvocation',
+  CapabilityDelegation = 'capabilityDelegation',
+  KeyAgreement = 'keyAgreement',
+}
+
+const TYPE_SIDETREE = 'sidetree';
+class VadeApiSidetree extends VadeApiShared {
```
Comment (wulfraem) [inline builds/wasm/src/vade/vade-api-sidetree.ts:33]: Could you add function docs? ^^

Also `createDid`'s function doc is rather brief regarding the arguments, and `getDid` documents more arguments than it has. ^^

**Review by wulfraem — CHANGES_REQUESTED** [review]

File: builds/wasm/src/vade/vade-api-bbs.ts (hunk 4 above)
Comment (wulfraem) [reply builds/wasm/src/vade/vade-api-bbs.ts:61]: Either GitHub is showing it wrong or it's still missing. Btw, "description" is missing as well. ^^

Comment (wulfraem) [reply builds/wasm/src/vade/vade-api-bbs.ts:61]: Also are the options in the function doc still up-to-date? Maybe search & replace all of them if need be. 🍡

**Review by wulfraem — CHANGES_REQUESTED** [review]

File: builds/wasm/src/vade/vade-api-bbs.ts (hunk 4 above)
Comment (S3bb1) [reply builds/wasm/src/vade/vade-api-bbs.ts:61]: 🤦  thx, updated

**Review by wulfraem — APPROVED** [review]

**Pull request merged**